pkg/executables/kubectl.go

```diff
@@ -1404,3 +1408,53 @@ func (k *Kubectl) GetResources(ctx context.Context, resourceType string, opts ..
 	stdOut, err := k.Execute(ctx, params...)
 	return stdOut.String(), err
 }
+
+// GetHardwareForCluster gets the hardwares with ownerName label.
+// Then iterates to filter the hardwares that correspond to the given clusterName.
+func (k *Kubectl) GetHardwareForCluster(ctx context.Context, clusterName, kubeconfigFile, namespace string) ([]tinkv1alpha1.Hardware, error) {
+	params := []string{
+		"get", captHardwareResourceType, "-o", "json", "--kubeconfig",
+		kubeconfigFile, "--namespace", namespace, "--selector=v1alpha1.tinkerbell.org/ownerName",
+	}
+	stdOut, err := k.Execute(ctx, params...)
+	if err != nil {
+		return nil, fmt.Errorf("getting hardware with selector: %v", err)
+	}
+
+	response := &tinkv1alpha1.HardwareList{}
+	err = json.Unmarshal(stdOut.Bytes(), response)
+	if err != nil {
+		return nil, fmt.Errorf("parsing get hardware response: %v", err)
+	}
+
+	// Filtering the hardware with ownerName mapping to cluster name.
+	var filteredHardwareList []tinkv1alpha1.Hardware
+	for _, hw := range response.Items {
+		if strings.Contains(hw.Labels["v1alpha1.tinkerbell.org/ownerName"], clusterName) {
+			filteredHardwareList = append(filteredHardwareList, hw)
+		}
+	}
+
+	return filteredHardwareList, nil
+}
+
+// ValidateBmcsPowerState validates if the given BMC CRs all have the desired powerState.
+func (k *Kubectl) ValidateBmcsPowerState(ctx context.Context, bmcNames []string, powerState, kubeconfigFile, namespace string) error {
```

Comment:
This is business logic - it shouldn't exist in the `Kubctl` executable. Why do this here instead of on a dedicated post delete validation construct within the tinkerbell provider package?

Reply:
Sure! I can remove the logic and make this method solely `kubectl get ...`.
And add a validation method that performs the checks.